### PR TITLE
chore: app type references added and type errors fixed

### PR DESCRIPTION
**Pull request by MEsteves22**
- As discussed I added the type references for the app to fix errors like this:

Screenshot 2023-06-21 at 15 30 59

- I also fixed type errors

### CODE REVIEW

File: app/src/components/components/BulkActions/BulkActions.tsx [hunk 1]
```diff
@@ -103,13 +103,21 @@ export const BulkActions = () => {
...
     newData[i + pageSize * page].checked = checked;
     setData(newData);
   };
 
-  const handleAction = (_, __, action: HvActionGeneric | HvListValue) => {
+  const handleAction = (
+    event: React.SyntheticEvent,
+    id: string,
+    action: HvActionGeneric | HvListValue
+  ) => {
```
Comment (zettca) [inline app/src/components/components/BulkActions/BulkActions.tsx:120]: We shouldn't be "redefining" the types. It's extra work, prone to errors, less future-proof, etc.

```suggestion
  const handleAction: HvBulkActions["actionsCallback"] = (event, id, action) => {
```

File: app/src/components/components/ButtonConfigurator/ButtonConfigurator.tsx [hunk 2]
```diff
@@ -35,7 +36,7 @@ export const ButtonConfigurator = () => {
           label: name,
         }))}
         onChange={(item) => {
-          setVariant(item.value as HvButtonVariant);
+          setVariant((item as HvListValue).value as HvButtonVariant);
```
Comment (zettca) [inline app/src/components/components/ButtonConfigurator/ButtonConfigurator.tsx:39]: We definitely need a proper Select element...

File: app/src/components/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -16,7 +16,14 @@ export const Calendar = () => {
...
 
-  const onChangeHandler = (event, value) => {
+  const onChangeHandler = (
+    event:
+      | React.ChangeEvent<
+          HTMLTextAreaElement | HTMLInputElement | HTMLButtonElement
+        >
+      | undefined,
+    value: Date | DateRangeProp
+  ) => {
```
Comment (zettca) [inline app/src/components/components/Calendar/Calendar.tsx:26]: same as above
```suggestion
  const onChangeHandler: HvCalendarProps["onChange"] = (event, value) => {
```

File: app/src/components/components/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -28,21 +35,23 @@ export const Calendar = () => {
...
+        endDate: value as Date,
       });
     }
   };
 
   const visibleDateChangeHandler = (
-    event: any,
+    event:
+      | React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+      | undefined,
```
Comment (zettca) [inline app/src/components/components/Calendar/Calendar.tsx:52]: same as above, and for other instances

**Review by zettca — APPROVED** [review]
Awesome 💪

Now we just need the same for our whole codebase 😅 

> TypeScript isn't making your life worse. It's just showing you how bad your life already is

File: app/src/components/components/ButtonConfigurator/ButtonConfigurator.tsx (hunk 2 above)
Comment (MEsteves22) [reply app/src/components/components/ButtonConfigurator/ButtonConfigurator.tsx:39]: Agreed 🙏

File: app/src/components/components/BulkActions/BulkActions.tsx (hunk 1 above)
Comment (MEsteves22) [reply app/src/components/components/BulkActions/BulkActions.tsx:120]: You're right, I will change this 🙏

**Review by plagoa — APPROVED** [review]

**Pull request merged**